jwql/jwql_monitors/monitor_cron_jobs.py

```diff
@@ -0,0 +1,311 @@
+    latest_time = os.path.getctime(latest)
+    return (latest, latest_time)
+
+
+def get_cadence(filenames):
+    """Calculate the cadence of the log files in a given directory.
+    Use timestamps
+
+    Parameters
+    ---------
+    filenames : list
+        List of log files to examine
+
+    Returns
+    -------
+    mean_delta : float
+        Mean time in seconds between the appearance of consecutive log files
+
+    stdev_delta : float
+        Standard deviation in seconds between the appearance of consecutive log files
+    """
+    minimum_log_num = 3  # Set to a low value for now since we don't have many logfiles
+    times = [os.path.getctime(filename) for filename in filenames]
+    if len(times) >= minimum_log_num:
+        sorted_times = np.array(sorted(times))
+        delta_times = sorted_times[1:] - sorted_times[0:-1]
+        mean_delta = np.mean(delta_times)
+        stdev_delta = np.std(delta_times)
+    else:
+        # If there are < minimum_log_num logfiles, then let's assume we can't
+        # get a reliable measure of cadence. Fall back to a value of
+        # 1 year between files, to avoid accidentally flagging this monitor
+        # as running late in the subsequent check
+        mean_delta = 31556736.0  # sec per year
+        stdev_delta = 31556736.0  # sec per year
+    return mean_delta, stdev_delta
+
+
+@log_fail
+@log_info
+def status(production_mode=True):
+    """Main function: determine the status of the instrument montiors by examining
+    log files.
+
+    Parameters
+    ----------
+    production_mode : bool
+        If true, look in the main log directory. If false, look in the dev log
+        file directory.
+
+    Returns
+    -------
+    logfile_status : dict
+        Nested dictionary containing the status for all monitors. Top level keys
+        include all monitors. Within a given monitor, the value is a dictionary
+        containing 'missing_file' and 'status' keys. 'missing_file' is a boolean
+        describing whether or not there is a suspected missing log file based
+        on the timestamps of the existing files. 'status' is a string that is
+        either 'success' or 'failure'.
+    """
+    # Begin logging
+    configure_logging('monitor_cron_jobs', production_mode=production_mode)
+    logging.info("Beginning cron job status monitor")
+
+    # Get main logfile path
+    settings = get_config()
+    log_path = settings['log_dir']
+
+    # If we are in development mode, the log files are in a slightly
+    # different location than in production mode
+    if not production_mode:
+        log_path = os.path.join(log_path, 'dev')
+
+    # Set up a dictionary to keep track of results
+    logfile_status = {}
+
+    # Get a list of the directories under the main logging directory.
+    generator = os.walk(log_path, topdown=True)
+
+    # Loop over monitors (note that subsubdir should always be empty)
+    for subdir, subsubdir, filenames in generator:
```

Comment:
@bhilbert4 I've noticed that if `log_path` is the one expected in `production_mode == True`, then this will include `dev/` logs, since `dev/` is a subdirectory.  You might want to add some code to make sure that subdirectory is not included in this `walk`.

Reply:
Done.